api/internals/usecase/festival_item_usecase.go

```diff
@@ -203,6 +206,148 @@ func (fiu *festivalItemUseCase) DestroyFestivalItem(c context.Context, id string
+			&festivalItemForMyPageColumn.IsSettled,
+		)
+		if err != nil {
+			fmt.Println(err)
+			return festivalItemDetailsList, err
+		}
+		festivalItemForMyPageColumns = append(festivalItemForMyPageColumns, festivalItemForMyPageColumn)
+	}
+
+	festivalItemDetailsList = convertColumnToGenerated(festivalItemForMyPageColumns)
+
+	return festivalItemDetailsList, nil
+}
+
 type FestivalItemDetails = generated.FestivalItemDetails
 type FestivalItem = generated.FestivalItem
 type FestivalItemWithBalance = generated.FestivalItemWithBalance
+type FestivalItemDetailsForMypage = generated.FestivalItemsForMyPage
+type FestivalItemWithReport = generated.FestivalItemWithReport
+type BuyReport = generated.BuyReportInformation
+
+func convertColumnToGenerated(festivalItemForMyPageColumns []domain.FestivalItemForMyPageColumn) []FestivalItemDetailsForMypage {
+	var festivalItemDetailsList []FestivalItemDetailsForMypage
+
+	// 部門ごとにマップを作成
+	var festivalItemDetailsForMypageMap = make(map[string]FestivalItemDetailsForMypage)
+	var festivalItemMaps = make(map[string]map[string]FestivalItemWithReport)
+
+	for _, festivalItemForMyPageColumn := range festivalItemForMyPageColumns {
+		festivalItemDetailsForMypage := festivalItemDetailsForMypageMap[festivalItemForMyPageColumn.DivisionName]
+		// 局と部門名前定義
+		festivalItemDetailsForMypage.DivisionName = &festivalItemForMyPageColumn.DivisionName
+		festivalItemDetailsForMypage.FinancialRecordName = &festivalItemForMyPageColumn.FinancialRecordName
+
+		// 予算と支出データ集計
+		festivalItemMap := festivalItemMaps[festivalItemForMyPageColumn.DivisionName]
+		if festivalItemMap == nil {
+			festivalItemMap = make(map[string]FestivalItemWithReport)
+		}
+		festivalItemWithReport := festivalItemMap[festivalItemForMyPageColumn.FestivalItemName]
+		festivalItemWithReport.FestivalItemName = &festivalItemForMyPageColumn.FestivalItemName
+
+		// totalがなければ定義
+		if festivalItemWithReport.FestivalItemTotal == nil {
+			expense, budget, balance := 0, 0, 0
+			var total Total
+			total.Expense, total.Budget, total.Balance = &expense, &budget, &balance
+			festivalItemWithReport.FestivalItemTotal = &total
+		}
+
+		*festivalItemWithReport.FestivalItemTotal.Budget += festivalItemForMyPageColumn.BudgetAmount
+		*festivalItemWithReport.FestivalItemTotal.Expense += festivalItemForMyPageColumn.ReportAmount
+		*festivalItemWithReport.FestivalItemTotal.Balance += festivalItemForMyPageColumn.BudgetAmount - festivalItemForMyPageColumn.ReportAmount
+
+		buyReports := festivalItemWithReport.BuyReports
+		if buyReports == nil {
+			var buyReportSlice []generated.BuyReportInformation
+			buyReports = &buyReportSlice
+		}
+
+		var buyReport BuyReport
+		buyReport.Id = &festivalItemForMyPageColumn.BuyReportId
+		buyReport.BuyReportName = &festivalItemForMyPageColumn.PaidBy
+		buyReport.Amount = &festivalItemForMyPageColumn.ReportAmount
+		buyReport.ReportDate = &festivalItemForMyPageColumn.ReportDate
+
+		if festivalItemForMyPageColumn.IsSettled {
```

Comment:
[imo]
switchの方がかっこいい！
```suggestion
		switch {
		case festivalItemForMyPageColumn.IsSettled:
				buyReport.Status = &isSettled
		case festivalItemForMyPageColumn.IsPacked:
				buyReport.Status = &isPacked
		default:
				buyReport.Status = &empty
		}
```

Reply:
個人的にはifの方が直感的な気もして、可読性は高そう
でもswitchの方が簡潔に書けるのでそっちにしました
[条件分岐をifに変更](https://github.com/NUTFes/FinanSu/pull/915/commits/9def7bafd8a85be62ed54b4106dd18f0d1a24a4c)